File: services/orchestrator/src/tools/pipeboard.py
"""Pipeboard MCP HTTP client."""

import os
from typing import Any

import httpx


async def call_tool(tool_name: str, arguments: dict[str, Any]) -> Any:
    base_url = os.getenv("PIPEBOARD_MCP_URL", "https://meta-ads.mcp.pipeboard.co")
    token = os.getenv("PIPEBOARD_TOKEN", "")

    async with httpx.AsyncClient(timeout=60.0) as client:
        response = await client.post(
            f"{base_url.rstrip('/')}/tools/{tool_name}",
            json=arguments,
            headers={"Authorization": f"Bearer {token}"} if token else {},
        )
        response.raise_for_status()
        return response.json()

File: services/worker/src/processors/pipeboard.ts
import axios from "axios"
import pino from "pino"

const logger = pino({ name: "processor:pipeboard" })

export async function executePipeboardWrite(
  actionType: string,
  payload: Record<string, unknown>,
) {
  if (process.env.WRITE_ENABLED !== "true") {
    throw new Error("WRITE_ENABLED is not true — Pipeboard writes blocked")
  }

  const url = process.env.PIPEBOARD_MCP_URL ?? "https://meta-ads.mcp.pipeboard.co"
  const response = await axios.post(`${url}/actions/${actionType}`, payload, {
    headers: {
      Authorization: `Bearer ${process.env.PIPEBOARD_TOKEN ?? ""}`,
    },
  })
  logger.info({ actionType, status: response.status }, "pipeboard_write_complete")
  return response.data
}
